### ISSUE TITLE
ac-html not removed.

### ISSUE THREAD

**Issue by popoleeMaster** [articodeltd/angular-cesium#306]
There was a ticket #275 and apparently it was fixed.

But using angular-cesium 0.60 I am doing the following :

TEMPLATE 

```
<ac-layer #menuLayer acFor="let menu of menu$" [context]="this" >
  <ac-html-desc props="{
    position: menu.position,
    fillColor : labelParam.fillColor,
    show: menu.show,
    horizontalOrigin : labelParam.horizontalOrigin,
    verticalOrigin : labelParam.verticalOrigin,
    scale: 0.1,
    pixelOffset : labelParam.pixelOffset,
  }">
    <ng-template let-menu>
      <div [hidden]="!menu.show" class="menu-container">
           my div is here
      </div>
    </ng-template>
  </ac-html-desc>
</ac-layer>
```

in component I have :

```import { Component, ViewChild, Input, AfterViewInit, ViewChildren, QueryList, ComponentRef, ChangeDetectorRef } from '@angular/core';
import { AcLayerComponent, AcNotification, LabelProps, ActionType } from 'angular-cesium';
import { Observable, Subject } from 'rxjs';
import { AppConstant } from 'src/app/app.constant';
import { v4 as uuid } from 'uuid';
import { AcMenuEntity } from 'src/app/utils/entities/ac-menu-entity';
import { DynamicComponent } from 'ng-dynamic-component';
import { Cartographic } from '../../models/navigation.models';
import { WAYPOINT_DRAWER_OPTIONS } from 'src/app/modules/navigation/navigation-route-style';


@Component({
  selector: 'app-entity-layer-menu',
  templateUrl: './entity-layer-menu.component.html',
  styleUrls: ['./entity-layer-menu.component.less']
})
export class EntityLayerMenuComponent implements AfterViewInit {
  @ViewChild('menuLayer') private menuLayer: AcLayerComponent;

  public menu$: Observable<AcNotification>;
  private menuEntity: AcMenuEntity;
  private menuEmitter: Subject<AcNotification> = new Subject<AcNotification>();

  public labelParam: LabelProps = {
    ...AppConstant.DEFAULT_LABEL_PROPS,
    pixelOffset : new Cesium.Cartesian2(0.0, -60)
  };

  constructor() {
    this.menu$ = this.menuEmitter.asObservable();
    this.menuEntity = new AcMenuEntity(
      uuid(),
      AppConstant.DEFAULT_POSITION,
      false
    );
  }

  ngAfterViewInit() {
    this.menuEmitter.next({
      id: this.menuEntity.id,
      actionType: ActionType.ADD_UPDATE,
      entity: this.menuEntity
    });
  }

  public show(position: Cartographic, posObs?: Observable<Cartographic>) {
    this.menuEntity.show = true;
    this.menuEntity.position = Cesium.Ellipsoid.WGS84.cartographicToCartesian(position);
    if (posObs) {
      posObs.subscribe(pos => {this.menuEntity.position = Cesium.Ellipsoid.WGS84.cartographicToCartesian(pos); this. refresh(); });
    }
    this.refresh();
  }

  public hide() {
    this.menuEntity.show = false;
    this.menuEntity.position = AppConstant.DEFAULT_POSITION;
    this.refresh();
  }

  private refresh() {
    this.menuEmitter.next({
      id: this.menuEntity.id,
      actionType: ActionType.ADD_UPDATE,
      entity: this.menuEntity
    });

/*
    this.menuLayer.update(this.menuEntity, this.menuEntity.id);
*/
  }
}
```

Result => 

https://prnt.sc/peouc9

The div should be a single one, but at the moment, every time I use refresh a new div is created.
I might do something wrong, but I do not know where cause I have a single ID and update an AcEntity with that id.

**Comment by popoleeMaster** [comment]
Doing this 

```
  private refresh() {
    this.menuLayer.remove(this.menuEntity.id);
    this.menuEmitter.next({
      id: this.menuEntity.id,
      actionType: ActionType.ADD_UPDATE,
      entity: this.menuEntity
    });
}
```

and even
```
  private refresh() {
    this.menuLayer.remove(this.menuEntity.id);
    this.menuLayer.update(this.menuEntity, this.menuEntity.id);
  }
```

"fix" the issue, but it imply recreating a DOM element every time.

The second exemple should actually not work since I removed the entity before requesting to update, but it does work.

**Comment by eitanfr** [comment]
I will check it out soon. 
thx

**Comment by eitanfr** [comment]
Hi, Fixed in #309 

Also, I see here you are trying to implement a context menu,
so check out [ContextMenuService](https://github.com/articodeltd/angular-cesium/blob/master/demo/app/components/context-menu-layer/context-menu-layer.component.ts) - similar to material dialog components, its create a component (your menu) and position it on the globe.
